src/java.base/share/classes/java/text/MessageFormat.java

```diff
@@ -1184,6 +1184,17 @@ public int hashCode() {
         return pattern.hashCode(); // enough for reasonable distribution
     }
 
+    /**
+     * {@return a string identifying this {@code MessageFormat}, for debugging}
+     */
+    @Override
+    public String toString() {
+        return
+            """
+            MessageFormat [locale: "%s", pattern: "%s"]
+            """.formatted(locale == null ? "None" : locale.getDisplayName(), toPattern());
```

Comment:
I think printing `null` for `locale == null` is better here, as `None` is sort of ambiguous. Applies to SDF (and related tests) as well.